modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeGridMonitorMonitorTest.java

```diff
@@ -0,0 +1,431 @@
+import org.apache.ignite.compute.ComputeJob;
+import org.apache.ignite.compute.ComputeJobAdapter;
+import org.apache.ignite.compute.ComputeJobResult;
+import org.apache.ignite.compute.ComputeTaskAdapter;
+import org.apache.ignite.compute.ComputeTaskFuture;
+import org.apache.ignite.compute.ComputeTaskSession;
+import org.apache.ignite.compute.ComputeTaskSessionFullSupport;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.failure.StopNodeFailureHandler;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.IgniteInterruptedCheckedException;
+import org.apache.ignite.internal.processors.task.monitor.ComputeGridMonitor;
+import org.apache.ignite.internal.processors.task.monitor.ComputeTaskStatusDiff;
+import org.apache.ignite.internal.processors.task.monitor.ComputeTaskStatusSnapshot;
+import org.apache.ignite.internal.util.future.GridFutureAdapter;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static org.apache.ignite.cluster.ClusterState.ACTIVE;
+import static org.apache.ignite.internal.processors.task.monitor.ComputeTaskStatusEnum.FAILED;
+import static org.apache.ignite.internal.processors.task.monitor.ComputeTaskStatusEnum.FINISHED;
+import static org.apache.ignite.internal.processors.task.monitor.ComputeTaskStatusEnum.RUNNING;
+import static org.apache.ignite.testframework.GridTestUtils.assertThrows;
+
+/**
+ * Test class for {@link ComputeGridMonitor}.
+ */
+public class ComputeGridMonitorMonitorTest extends GridCommonAbstractTest {
+    /** Coordinator. */
+    private static IgniteEx CRD;
+
+    /** Compute task status monitor. */
+    private ComputeGridMonitorImpl monitor;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        stopAllGrids();
+
+        IgniteEx crd = startGrids(2);
+
+        crd.cluster().state(ACTIVE);
+
+        awaitPartitionMapExchange();
+
+        CRD = crd;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        super.afterTestsStopped();
+
+        stopAllGrids();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        CRD.context().task().listenStatusUpdates(monitor = new ComputeGridMonitorImpl());
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        CRD.context().task().stopListenStatusUpdates(monitor);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
```

Comment:
I'm pretty sure there's a "getFailureHandler" method in base test class. You could just override it instead.